bin/hdfcoinc/pycbc_multiifo_dtphase

```diff
@@ -0,0 +1,169 @@
+parser.add_argument('--relative-sensitivities', nargs='+', type=float)
+parser.add_argument('--seed', type=int, default=124)
+parser.add_argument('--output-file')
+parser.add_argument('--verbose', action='store_true')
+parser.add_argument('--smoothing-sigma', type=int, default=2)
+parser.add_argument('--timing-uncertainty', type=float, default=.001,
+                    help="Timing uncertainty to set bin size and smoothing interval")
+parser.add_argument('--phase-uncertainty', type=float, default=0.25,
+                    help="Phase uncertainty used to set bin size and smoothing")
+parser.add_argument('--snr-reference', type=float, default=5, 
+                    help="Reference SNR to scale SNR uncertainty")
+parser.add_argument('--snr-uncertainty', type=float, default=1.4, 
+                    help="SNR uncertainty to set bin size and smoothing")
+args = parser.parse_args()
+
+twidth = args.timing_uncertainty  # approximate timing error at lower SNRs
+pwidth = args.phase_uncertainty   # approximate phase error at lower SNRs
+serr = args.snr_uncertainty   #
+sref = args.snr_reference     # Reference SNR for bin smoothing
+swidth = serr / sref
+
+srbmax = int((args.snr_ratio / swidth))
+srbmin = int((1.0 / args.snr_ratio) / swidth)
+
+# Apply a simple smoothing to help account for
+# measurement errors for weak signals
+def smooth_param(data, sigma, index):
+    bins = numpy.arange(-args.smoothing_sigma, args.smoothing_sigma+1)
+    kernel = norm.pdf(bins)
+    
+    nweights = {}
+    
+    # This is massively redundant as many points may be 
+    # recalculated, but it's straightforward.
+    for key in weights:
+        for a in bins:
+            nkey = list(key)
+            nkey[index] += a
+            
+            weight = 0
+            for b, w in zip(bins, kernel):
+                wkey = list(nkey)
+                wkey[index] += b
+                wkey = tuple(wkey)
+                
+                if wkey in data:
+                    weight += data[wkey] * w
+                
+            nweights[tuple(nkey)] = weight
+            
+    return nweights
+    
+d = {ifo: pycbc.detector.Detector(ifo) for ifo in args.ifos}
+
+pycbc.init_logging(args.verbose)
+
+numpy.random.seed(args.seed)
+size = 1000000
+chunks = int(args.sample_size / size) + 1
+
+l = 0
+nsamples = 0
+weights = {}
+for k in range(chunks):
+    nsamples += size
+    logging.info('generating %s samples' % size)
+
+    # Choose random sky location and polarizations from
+    # an isotropic population
+    ra = uniform(0, 2 * numpy.pi, size=size)
+    dec = numpy.arccos(uniform(-1., 1., size=size)) - numpy.pi/2
+    inc = numpy.arccos(uniform(-1., 1., size=size))
+    pol = uniform(0, 2 * numpy.pi, size=size)
+    ip = numpy.cos(inc)
+    ic = 0.5 * (1.0 + ip * ip)
+
+    # calculate the toa, poa, and amplitude of each sample
+    data = {}
+    for r, ifo in enumerate(args.ifos):
```

Comment:
would be simpler to cycle over `rs, ifo in zip(args.relative_sensitivities, args.relative_ifos)`, the way the relative sensitivities are accessed atm obscures what is going on

Reply:
Yep, that makes sense too.